### PR TITLE
code added

### PR DESCRIPTION
**Pull request by anshmishra010**
Code added in C++ language in code folder .
Wanted to add code as a GSSoC'21 participant .

### CODE REVIEW

**Comment by rudrakshi99** [comment]
link it with the issue.

**Comment by anshmishra010** [comment]
#1660

File: Code/C++/palindrome_number.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+{
+    int num;
+    cout << "Enter the number to check : " << endl;
+    cin >> num;
+    checkPalindrome(n) ? cout << "True" : cout << "False";
+    return 0;
+}
+
+/*
+Time complexity : O(n2)
```
Comment (rish-singhal) [inline Code/C++/palindrome_number.cpp:43]: what is n2?? is it n^2? and explanation? also what is n?

File: Code/C++/palindrome_number.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+{
+    // divi =  divisor
+    int divi = 1;
+
+    //finding the apt. divisor for the number
+    while (num / divi >= 10)
+    {
+        divi *= 10;
+    }
+    while (num != 0)
```
Comment (rish-singhal) [inline Code/C++/palindrome_number.cpp:19]: add extra line

File: Code/C++/palindrome_number.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/*       Palindrome Number : Given an integer x, return true if x is palindrome integer.
+                            An integer is a palindrome when it reads the same backward as forward.
+                            This code is implemented without using extra space.
```
Comment (rish-singhal) [inline Code/C++/palindrome_number.cpp:3]: + add algorithm too!

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: Code/C++/palindrome_number.cpp (hunk 1 above)
Comment (anshmishra010) [reply Code/C++/palindrome_number.cpp:43]: O(n*n)

Comment (anshmishra010) [reply Code/C++/palindrome_number.cpp:43]: n =  number of elements

Comment (rish-singhal) [reply Code/C++/palindrome_number.cpp:43]: update it in your next commit

Comment (anshmishra010) [reply Code/C++/palindrome_number.cpp:43]: yes

Comment (anshmishra010) [reply Code/C++/palindrome_number.cpp:43]: should i do any more changes or it's fine

Comment (rish-singhal) [reply Code/C++/palindrome_number.cpp:43]: update in the README.md too with the link to this code. Everything else seems fine

File: Code/C++/palindrome_number.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    return 0;
+}
+
+/*
+Time complexity : O(n2)
+Space complexity : O(1)
+*/
+
+/*
+Test Case :
```
Comment (rish-singhal) [inline Code/C++/palindrome_number.cpp:48]: update to Test Case"s"

File: Code/C++/palindrome_number.cpp (hunk 1 above)
Comment (rish-singhal) [reply Code/C++/palindrome_number.cpp:43]: and add explanation why it is O(n^2)

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by anshmishra010** [comment]
Please check now commits

File: Code/C++/palindrome_number.cpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+{
+    int num;
+    cout << "Enter the number to check : " << endl;
+    cin >> num;
+    checkPalindrome(n) ? cout << "True" : cout << "False";
+    return 0;
+}
+
+/*
+Time complexity : O(log n)
```
Comment (rish-singhal) [inline Code/C++/palindrome_number.cpp:45]: what is n, define here

File: Code/C++/palindrome_number.cpp (hunk 4 above)
Comment (rish-singhal) [reply Code/C++/palindrome_number.cpp:50]: still not resolved

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by rish-singhal** [comment]
Also update the link to the code in the readme

File: Code/C++/palindrome_number.cpp (hunk 3 above)
Comment (anshmishra010) [reply Code/C++/palindrome_number.cpp]: @rish-singhal  it's just a mathematical type logic ,so it doesn't include any famous algo . Still I have added the description of the code .

**Comment by anshmishra010** [comment]
Please check now
And as I new to algo-tree so I am unable to find the readme to update this code .
So please if possible send the link of readme so that I can update there too

File: Code/C++/palindrome_number.cpp (hunk 3 above)
Comment (rish-singhal) [reply Code/C++/palindrome_number.cpp]: What are you doing to check if it is palindrome, add the steps.

**Comment by rish-singhal** [comment]
PR without assignment.

**Comment by anshmishra010** [comment]
Why it's invalid ? @rish-singhal 
I have attached the issue number #1660 .

**Comment by rish-singhal** [comment]
PRs are invalid if the issue is not assigned to the one who makes the PR. Wait first the issue is assigned to you.

**Comment by anshmishra010** [comment]
So any more required changes ?

**Comment by rish-singhal** [comment]
Readme file is in array/number theory

**Comment by anshmishra010** [comment]
Updated readme

**Comment by anshmishra010** [comment]
@rish-singhal is there any more changes to do ? If no then please merge it .

**Comment by rish-singhal** [comment]
There are changes, so look at the comments

**Comment by anshmishra010** [comment]
-> I have added "s"  in test case .
-> I have changed the time complexity too.
-> I have updated the readme too .
anything else @rish-singhal ,

**Comment by rish-singhal** [comment]
1. Add information about the method you are using. For example, reversing the digits or something like that.

2. The readme file you updated is incorrect.

3. There should be test cases not test case's

**Comment by anshmishra010** [comment]
added a short description about the algo , updated readme , added "s" 
@rish-singhal

**Comment by rish-singhal** [comment]
> Readme file is in algorithm/number theory 

This is the readme file you have to update.

**Comment by rish-singhal** [comment]
And according to your description, it comes out that if the first and last points are not same then return false,

but what about 1006541?

- you should also mention that you are updating those points

**Comment by anshmishra010** [comment]
Done @Rohan27s

**Comment by rish-singhal** [comment]
Just update the spelling of satisfied in the description*

**Comment by rish-singhal** [comment]
And also, sort the list according to the name of the algorithm in readme.

**Comment by anshmishra010** [comment]
changed spelling

**Comment by anshmishra010** [comment]
@rish-singhal

**Comment by rish-singhal** [comment]
sort the list according to the name of the algorithm in readme

**Comment by anshmishra010** [comment]
Screenshot (150)

Currently this is the readme , just tell me where should I place it .

**Comment by anshmishra010** [comment]
@rish-singhal

**Comment by rish-singhal** [comment]
It's okay to place in fundamental, just sort the list -- 
The links should be lexicographic order.


A, B, C,.... 

https://en.m.wikipedia.org/wiki/Lexicographic_order

**Comment by rish-singhal** [comment]
image

**Comment by anshmishra010** [comment]
Screenshot (151)
Now fine ? Should I push it

**Comment by rish-singhal** [comment]
Just switch Fibonacci and factorial

Fa.. comes before Fi..

Then push the changes

**Comment by anshmishra010** [comment]
@rish-singhal

**Comment by rish-singhal** [comment]
great!

**Comment by anshmishra010** [comment]
@rish-singhal Though I know my code is not that great , but Level 0 it too low .
Usually in other projects mentors are giving the Level  0 for just a file deletion or like that , so I request you to add level 1 tag at least .

**Comment by rish-singhal** [comment]
I don't have a say in this... this is what the group follows.

As the code is simply iteration over digits of a number. And does not require any extra data structure or algorithm: Level0 is assigned.

You can ask @rudrakshi99 in case you have doubt, if she's okay with it. She will assign level1.

I am just following the instructions.

**Comment by anshmishra010** [comment]
Okay , I agree to your point of view .
But it's a request so if possible then please add it  else I can't argue .
@rudrakshi99 @rish-singhal

**Pull request merged**